lib/js/jsfetch.nim

```diff
@@ -0,0 +1,90 @@
+## - Fetch for the JavaScript target: https://developer.mozilla.org/docs/Web/API/Fetch_API
+from asyncjs import PromiseJs
+from httpcore import HttpMethod
+export PromiseJs, HttpMethod
+
+when not defined(js) and not defined(nimdoc):
+  {.fatal: "Module jsfetch is designed to be used with the JavaScript backend.".}
+when defined(nodejs):
+  {.warning: "By design 'fetch()' is defined on the web browser, but may not be on NodeJS.".}
+
+type
+  FetchOptions* = ref object
+    `method`, body, integrity, referrer, mode, credentials, cache, redirect, referrerPolicy: cstring
```

Comment:
instead of quoting "method" (painful to use), how about something like (untested):
```nim
type
  FetchOptions* = ref object
    metod {.importc: "method".}: cstring
    body, integrity ...
```
(using `metod` as you did below, which is a good enough name)